consensus/src/main/kotlin/maru/consensus/validation/BlockValidator.kt

```diff
@@ -19,19 +19,31 @@ import com.github.michaelbull.result.Err
 import com.github.michaelbull.result.Ok
 import com.github.michaelbull.result.Result
 import encodeHex
+import java.util.Optional
+import maru.consensus.ProposerSelector
+import maru.consensus.ValidatorProvider
+import maru.consensus.qbft.adapters.toBeaconBlock
+import maru.consensus.state.StateTransition
+import maru.consensus.toConsensusRoundIdentifier
+import maru.consensus.validation.BlockValidator.BlockValidationError
 import maru.consensus.validation.BlockValidator.Companion.error
 import maru.consensus.validation.BlockValidator.Companion.ok
 import maru.core.BeaconBlock
 import maru.core.BeaconBlockHeader
+import maru.core.BeaconState
 import maru.core.HashUtil
 import maru.core.Validator
+import maru.database.BeaconChain
 import maru.executionlayer.client.ExecutionLayerClient
 import maru.executionlayer.extensions.hasValidExecutionPayload
 import maru.serialization.rlp.bodyRoot
+import maru.serialization.rlp.stateRoot
 import org.hyperledger.besu.consensus.common.bft.BftHelpers
+import org.hyperledger.besu.consensus.qbft.core.types.QbftBlock
+import org.hyperledger.besu.consensus.qbft.core.types.QbftBlockValidator
 import tech.pegasys.teku.infrastructure.async.SafeFuture
 
-fun interface BlockValidator {
+interface BlockValidator : QbftBlockValidator {
```

Comment:
This is a negative change IMO for 3 reasons:
* it couples `BlockValidator` to Besu's internals. The inital approach with independent `BlockValidator` and `QbftBlockValidatorAdapter` was cleaner and it 
* Now this interface violates single responsibility principle as we have to implement both `validateBlock` and `validateBlock`. Having both is also confusing for the reader IMO
* We can't make it a functional interface now, which will make the tests that are using `BlockValidator` more verbose

I don't really see the benefit of mixing `QbftBlockValidator` and `BlockValidator` into the same interface other than removal of `QbftBlockValidatorAdapter` which would be trivial anyway